tests/test_encode_decode.py

```diff
@@ -263,3 +266,22 @@ def test_quopri_encode_decode_round_trip(self, payload, quotetabs, header):
         encoded = quopri.encodestring(payload, quotetabs=quotetabs, header=header)
         decoded = quopri.decodestring(encoded, header=header)
         self.assertEqual(payload, decoded)
+
+
+class TestBinhex(unittest.TestCase):
+    @given(payload=st.binary())
+    def test_binhex_encode_decode(self, payload):
+        with TemporaryDirectory() as dirname:
+            input_file_name = os.path.join(dirname, "input.txt")
+            encoded_file_name = os.path.join(dirname, "encoded.hqx")
+            decoded_file_name = os.path.join(dirname, "decoded.txt")
+            with open(input_file_name, "wb", buffering=0) as input_file:
+                input_file.write(payload)
+            binhex.binhex(input_file_name, encoded_file_name)
+            binhex.hexbin(encoded_file_name, decoded_file_name)
+            with open(decoded_file_name, "rb", buffering=0) as decoded_file:
+                decoded_payload = decoded_file.read()
+            assert payload == decoded_payload
+            os.remove(input_file_name)
+            os.remove(encoded_file_name)
+            os.remove(decoded_file_name)
```

Comment:
Last change, I promise: because the `TemporaryDirectory()` is deleted when the context manager closes, we don't need to explicitly remove these files.

We also don't need to pass `buffering=0` above; the default behaviour of `open()` is fine.

Thanks for your patience - I'll merge after this!

Reply:
Done 😄